docs/virtualkeyboard/index.html

```diff
@@ -0,0 +1,492 @@
+    <title>VirtualKeyboard API</title>
+    <script src='https://www.w3.org/Tools/respec/respec-w3c-common' defer class='remove'></script>
+    <script class='remove'>
+        var respecConfig = {
+              specStatus:   "ED"
+          ,   shortName:    "virtualKeyboard"
+          ,   editors:      [{ name: "Anupam Snigdha",
+                               mailto: "snianu@microsoft.com",
+                               company: "Microsoft Corporation"},
+                               { name: "Grisha Lyukshin",
+                               mailto: "glyuk@microsoft.com",
+                               company: "Microsoft Corporation"},]
+          ,   license:      "w3c-software-doc"
+          ,   wgPublicList: "public-editing-tf"
+	  ,   otherLinks: [{
+                  key: 'Participate',
+                  data: [{
+                      value: 'We are on GitHub.',
+                      href: 'https://github.com/w3c/editing'
+                  }, {
+                      value: 'File a bug.',
+                      href: 'https://github.com/w3c/editing/issues'
+                  }, {
+                      value: 'Commit history.',
+                      href: 'https://github.com/w3c/editing/commits/gh-pages'
+                  }, {
+                      value: 'Mailing list.',
+                      href: 'http://lists.w3.org/Archives/Public/public-editing-tf/'
+                  }]
+              }]
+          ,   wg: "Web Applications Working Group"
+          ,   wgURI: "https://www.w3.org/2019/webapps/"
+          ,   wgPatentURI: "https://www.w3.org/2004/01/pp-impl/114929/status"
+          ,   edDraftURI:   "http://w3c.github.io/editing/docs/virtualKeyboard/"
+          , xref: ["WebIDL"
+          , "DOM"
+          , "HTML"
+          , "geometry"]
+        };
+    </script>
+    <script src='https://www.w3.org/Tools/respec/respec-w3c' class='remove'></script>
+</head>
+
+<body>
+    <section id='abstract'>
+        <p>
+            The {{VirtualKeyboard}} API provides web developers a more granular control of on-screen keyboard (virtual keyboard or VK). The API also provides information about VK geometry, such as rect occlusion and gives ability to opt-out from resizing of the visual viewport in the browser.
+        </p>
+    </section>
+    <section id='sotd'>
+      <p>
+      </p>
+    </section>
+    <section>
+        <h2>Introduction</h2>
+        <P>This section is non-normative.</P>
+        <p>
+            The Virtual Keyboard (VK) is the on-screen keyboard used for input in scenarios where a hardware keyboard may not be available. Unlike hardware keyboard, VK can adapt its shape to optimize for the expected type of input.
+            In addition, user agents respond to the presence of the VK, without any exposure of this information to web developers. This document proposes a new web API surface for VK that developers can use to enable better customization of their webpage's content and experiences.
```

Comment:
I think it would be useful in this paragraph to expand on how user agents currently respond to the presence of the VK and what alternatives this API offers.  As is I don't think it adds anything compared to the abstract.

Maybe:  User agents respond to the presence of the VK by:
1. Repositioning the user agent above the VK
2. Reducing the size of the layout viewport so the VK doesn't occlude it
3. Reducing the size of the visual viewport and padding the layout viewport to ensure it can be shifted above the VK

This API provides a fourth option that allows the user agent to leave its layout and visual viewports unchanged and instead provide information about the intersection of the VK and layout viewport so that the author can adapt the layout of their web pages using JavaScript or CSS environment variables.

It would be good to have graphics illustrating the difference between these behaviors.

Reply:
updated